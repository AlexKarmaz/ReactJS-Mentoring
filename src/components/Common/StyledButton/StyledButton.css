.styledButton{
    cursor: pointer;
    text-transform: uppercase;
    font-weight: 100;
    width: fit-content;
    border-color: transparent;
}

.styledButton.confirm{
    color: #ffffff;
    background-color: #f65261;
}

.styledButton.action{
    color: #F65261;
    background-color: #636262d1
}

.styledButton.medium{
    font-size: 20px;
    padding: 10 20;
}
