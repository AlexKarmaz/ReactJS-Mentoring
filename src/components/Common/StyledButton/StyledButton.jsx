import React from 'react';
import PropTypes from 'prop-types';
import './StyledButton.css';

const StyledButton = ({size='medium', onClick, text, type='confirm', className}) => (
    <button type='button' className={`styledButton ${className} ${size} ${type}`} onClick={onClick}>
        {text}
    </button>
);

StyledButton.propTypes = {
    /**
     * How large should the button be?
     */
    size: PropTypes.string,
    /**
     * Click handler
     */
    onClick: PropTypes.func.isRequired,
    /**
     * Text for the button
     */
    text: PropTypes.string,
    /**
     * Button type
     */
    type: PropTypes.string,
    /**
     * CSS class
     */
    className: PropTypes.string,
};

StyledButton.defaultProps = {
    size: 'medium',
    text: 'button',
    className: 'className'
};

export default StyledButton;
