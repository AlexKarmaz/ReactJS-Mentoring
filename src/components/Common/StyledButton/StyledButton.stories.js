/* eslint-disable*/
import React from 'react';
import { withKnobs, text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import StyledButton from './StyledButton';

const wrapperStyle = {
  backgroundColor: '#232323',
};

export default {
  title: 'Styled Button',
  component: StyledButton,
  decorators: [withKnobs],
};

export const Default = () => (
  <div  style={wrapperStyle}>
    <StyledButton
      onClick={action('StyledButton-click')}
      text={text("Text","Styled button")}

    />
  </div>
);

export const Reset = () => (
  <div  style={wrapperStyle}>
    <StyledButton
      onClick={action('StyledButton-click')}
      text={text("Text","Styled button")}
      type="reset"
    />
  </div>
);

export const Action = () => (
  <div  style={wrapperStyle}>
    <StyledButton
      onClick={action('StyledButton-click')}
      text={text("Text","Styled button")}
      type="action"
    />
  </div>
);

