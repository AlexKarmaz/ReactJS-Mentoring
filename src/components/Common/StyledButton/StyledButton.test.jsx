import { shallow } from 'enzyme';
import React from 'react';
import StyledButton from './StyledButton';

describe('StyledButton', () => {
    it('should be rendered', () => {
        const onClick = jest.fn();

        const styledButton = shallow(<StyledButton onClick={onClick}/>);

        expect(styledButton).toMatchSnapshot();
    });

    it('should receive correct props', () => {
        const onClick = jest.fn();

        const styledButton = shallow(<StyledButton onClick={onClick} text={'test'} />);
        styledButton.simulate('click');

        expect(styledButton.find('button').hasClass('medium confirm')).toBeTruthy();
        expect(styledButton.find('button').text()).toEqual('test');
        expect(onClick).toBeCalled();
    });
});
