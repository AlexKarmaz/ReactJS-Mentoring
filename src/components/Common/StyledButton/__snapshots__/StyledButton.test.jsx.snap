// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton should be rendered 1`] = `ShallowWrapper {}`;
